core/server.go

```diff
@@ -163,6 +164,10 @@ func (s *Server) Serve(ctx context.Context, conn net.PacketConn) error {
 				}
 				defer stream.Close()
 
+				if ok := s.handshakeWithtimeout(conn, stream, 10*time.Second); !ok {
+					return
```

Comment:
Suggest close connection with an error msg of handshake timeout

Reply:
or, a `GoawayFrame`?

Reply:
The code that response `goaway` or `ack` is placed in `handshake` function scope.